In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data. Change winner to binary 1/0:
load( '../out/d3-fight-level-standardize-normalize.pkl' )
y[ y == -1 ] = 0

In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [3]:
# Fit decision tree.
from sklearn.tree import DecisionTreeClassifier
m = DecisionTreeClassifier( max_depth = 14, min_samples_split = .05, random_state = 716 )
m.fit( X_train, y_train )

# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_train, m.predict(X_train) ) )

              precision    recall  f1-score   support

           0       0.70      0.60      0.65       636
           1       0.79      0.85      0.82      1099

    accuracy                           0.76      1735
   macro avg       0.74      0.73      0.73      1735
weighted avg       0.76      0.76      0.76      1735



In [4]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_test, m.predict(X_test) ) )

              precision    recall  f1-score   support

           0       0.50      0.44      0.47       215
           1       0.69      0.74      0.72       364

    accuracy                           0.63       579
   macro avg       0.60      0.59      0.59       579
weighted avg       0.62      0.63      0.62       579



f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).